### DISCUSSION TITLE
positional argument position disparity between pyEMS and ERL/EMS

### DISCUSSION THREAD

**Discussion by TfedUD** [NREL/EnergyPlus#9020] [Q&A]
Hello!

This may just be my OCD, and I would be more than thrilled to fork and PR EnergyPlus (would be a huge happy dance for me)
!!!! In terms of the Py side; c++ is NOT something I know how to write !!!!!

But the positional argument sequence between, for instance:

```idf
EnergyManagementSystem:Sensor,
  Whtwovol,                                      !- Name
  Main AHU AHU Heat Recovery Relief Outlet,      !- Output:Variable or Output:Meter Index Key Name
  System Node Standard Density Volume Flow Rate; !- Output:Variable or Output:Meter Name
```  
and for instance:
```python
self.handles["Whtwovol"] = self.api.exchange.get_variable_handle(state,
                                                                 "System Node Standard Density Volume Flow Rate",
                                                                 "Main AHU AHU Heat Recovery Relief Outlet")
```

**The inverted nature of:**
```
Output:Variable or Output:Meter Index Key Name
Output:Variable or Output:Meter Name
```
in native EMS and 
```python
"""Python EMS"""
def func(state, _meter_name,
                _meter_index_key_name):
```
between pyEMS and native EMS; 
while understanding that the 'state being requested is the first argument following ```state```, is the "state being requested".

I am curious as to the possibility in terms of user-friendliness, and a bit of standardization across the board between the two, in terms of the positional arguments; that may aide the user in switching back and forth between the two?  one less item to hold in 'grey sponge RAM'

I love E+, thanks to everyone who has been maintaining and contributing; I hope to contribute some day myself!
( a dark mode option for EP-launch may likely; sans this may well be my first PR :sweat_smile: ) 

-TF

**Comment by bonnema** [comment]
I noticed this very same thing. I think there is a method to @Myoldmopar's madness :grin: and having the order match is a non-trivial change. I am not sure the exact reasons, maybe @Myoldmopar or @mitchute can shed some more light...

**Comment by mitchute** [comment]
Not I... I doubt it's actually that difficult of a change to make happen, but I bet there's a reason for it being the way it is.

**Reply by TfedUD** [reply]
I plan on taking a deeper look at the src over the weekend, I've unfortunately not had time to just yet. good fun! (Not saying I think I for sure will be able to ID but good learning)

**Comment by Myoldmopar** [comment]
Unless someone is also planning on setting up an automatic conversion of every user's Python API scripts to go to the new argument order, then I highly anticipate we will not be breaking our API for this.

**Reply by TfedUD** [reply]
This is a fantastic learning point for myself on maintaining software, thank you!

**Reply by Myoldmopar** [reply]
I do appreciate your interest, and I don't want to stifle it!  But each release we spend a chunk of time just producing tools to update user files from one version from another.  The more things we break from one version to the next, the more we have to maintain :sweat_smile:

**Reply by TfedUD** [reply]
No stifling at all! Learning is fun, just happy to be in a discussion about all this!
